### PR TITLE
enhancement(helm platform): Add DNS Policy and Config, Liveness and Readyness probes.

### PR DESCRIPTION
**Pull request by MonolithProjects**
Hi there. This PR is adding options to the vector-agent helm chart to:

- set liveness probe
- set readyness probe
- set dnsPolicy
- add dnsConfig

### CODE REVIEW

File: distribution/helm/vector-agent/values.yaml [hunk 1]
```diff
@@ -32,6 +32,33 @@ updateStrategy: RollingUpdate
 # Only used when updateStrategy is set to "RollingUpdate".
 maxUnavailable: 1
 
+# Use Liveness Probe
```
Comment (spencergilbert) [inline distribution/helm/vector-agent/values.yaml:35]: One thing we may want to add as a comment is that this isn't available by default, and the user needs to enable the `api` themselves to get this endpoint exposed.

File: distribution/helm/vector-agent/values.yaml [hunk 2]
```diff
@@ -32,6 +32,33 @@ updateStrategy: RollingUpdate
...
+livenessProbe: {}
+  # httpGet:
+  #   path: /health
+  #   port: api
+
+# Use Readyness Probe
```
Comment (spencergilbert) [inline distribution/helm/vector-agent/values.yaml:41]: One thing we may want to add as a comment is that this isn't available by default, and the user needs to enable the `api` themselves to get this endpoint exposed.

**Review by spencergilbert — COMMENTED** [review]
Thanks for the PR! I left one comment around documentation which I think could be useful for end users

File: distribution/helm/vector-agent/values.yaml (hunk 2 above)
Comment (MonolithProjects) [reply distribution/helm/vector-agent/values.yaml]: Good point. Added...

**Review by spencergilbert — APPROVED** [review]
Thanks again - is there any reason to not add this to the aggregator chart as well, I can't think of one?

**Comment by MonolithProjects** [comment]
Hmm i don't think so. I will add it there too...

**Comment by MonolithProjects** [comment]
Added also to the vector-aggregator helm chart

**Review by spencergilbert — APPROVED** [review]
Thanks so much!

**Pull request merged**